resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientConfigProviderFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.jboss.resteasy.client.jaxrs.internal;
+
+import org.jboss.resteasy.client.jaxrs.spi.ClientConfigProvider;
+import org.jboss.resteasy.util.BasicAuthHelper;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Client filter that will attach authorization header with either HTTP Basic auth or Bearer token auth.
+ * Credentials and token are loaded from ClientConfigProvider implementation.
+ *
+ * @author dvilkola@redhat.com
+ */
+public class ClientConfigProviderFilter implements ClientRequestFilter {
+
+    private final ClientConfigProvider clientConfigProvider;
+
+    public ClientConfigProviderFilter(final ClientConfigProvider clientConfigProvider) {
+        this.clientConfigProvider = clientConfigProvider;
+    }
+
+    @Override
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        if (requestContext.getHeaderString(HttpHeaders.AUTHORIZATION) == null) {
+            URI uri = requestContext.getUri();
+            String token = clientConfigProvider.getBearerToken(uri);
```

Comment:
ClientConfigProvider doesn't have any javadoc that guarantees that it will behave nicely if uri is null. Could there be a circumstance in which url == null? Maybe this is a stretch, but what if an application has a ClientRequestFilter that supplies the target URL, and that filter runs after this one? I don't know. But maybe ClientConfigProvider just needs some javadoc that defines its behavior. WDYT, @Skyllarr ?

Reply:
@ronsigal Thank you for review, this is a very good point. I thought that requestContext always contains uri that is not null when filters are being evaluated, but that was probably wrong assumption. We could throw an exception of continue silently if uri is null here. 

I think it is best to log it as warning and continue silently to avoid problems in case elytron configuration is present on classpath for other clients that use it.